src/structlog/stdlib.py

```diff
@@ -664,6 +667,66 @@ def add_logger_name(
     return event_dict
 
 
+_LOG_RECORD_KEYS = logging.LogRecord(
+    "name", 0, "pathname", 0, "msg", tuple(), None
+).__dict__.keys()
+
+
+class ExtraAdder:
+    """
+    Add extra attributes of `logging.LogRecord` objects to the event
+    dictionary.
+
+    This processor can be used for adding data passed in the ``extra``
+    parameter of the `logging` module's log methods to the event dictionary.
+
+    :param allow: An optional collection of attributes that, if present in
+        `logging.LogRecord` objects, will be copied to event dictionaries.
+
+        If ``allow`` is None all attributes of `logging.LogRecord` objects that
+        do not exist on a standard `logging.LogRecord` object will be copied to
+        event dictionaries.
+
+    .. versionadded:: 21.5.0
+    """
+
+    __slots__ = ["_copier"]
+
+    def __init__(self, allow: Optional[Collection[str]] = None) -> None:
+        self._copier: Callable[[EventDict, logging.LogRecord], None]
+        if allow is not None:
+            self._copier = functools.partial(self._copy_allowed, [*allow])
```

Comment:
Given that `_copy_allowed` takes `Collection[str]` too, this conversion shouldn't be necessary, right?

```suggestion
            self._copier = functools.partial(self._copy_allowed, allow)
```

Reply:
(i'll leave it in, since the copy can save some surprises…)